OpenRA.Game/Graphics/TerrainSpriteLayer.cs

```diff
@@ -12,12 +12,15 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Runtime.CompilerServices;
 
 namespace OpenRA.Graphics
 {
 	public sealed class TerrainSpriteLayer : IDisposable
 	{
-		static readonly int[] CornerVertexMap = { 0, 1, 2, 2, 3, 0 };
+		// PERF: we can reuse the IndexBuffer as all layers have the same size.
+		static readonly ConditionalWeakTable<World, IndexBufferRc> IndexBuffers = new();
+		readonly IndexBufferRc indexBuffer;
```

Comment:
Might be worth calling this `bufferWrapper` or something just to indicate it's not the actual index buffer.